### PR TITLE
Adds jupyter-ai and jupyter-ai-magics YAML files

### PR DESCRIPTION
**Pull request by JasonWeill**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- N/A If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/jupyter-ai-magics/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - hatchling >=1.4.0
+    - hatch-nodejs-version
+    - pip
```
Comment (3coins) [inline recipes/jupyter-ai-magics/meta.yaml:22]:
```suggestion
    - pip
    - hatch-jupyter-builder
```

**Comment by andrii-i** [comment]
I agree to be a maintainer

**Comment by dlqqq** [comment]
I agree to be a maintainer.

**Comment by 3coins** [comment]
I agree to be a maintainer.

**Comment by fcollonval** [comment]
Ok looking at the `faiss-cpu` issue, as the conda package does not install a empty Python package shell as PyPI does. A _dirty_ solution could be to pip install in the test a empty `faiss-cpu` Python package.

**Comment by 3coins** [comment]
@fcollonval 
Thanks for looking into this. I rebased and added `pip install faiss-cpu`; monitoring the CI if this works.

**Comment by 3coins** [comment]
@conda-forge-admin, please ping team. PR is ready for review.

**Comment by JasonWeill** [comment]
@conda-forge/help-python Ready for review.

File: recipes/jupyter-ai/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - jupyterlab >=4.0,<5
+    - hatch-nodejs-version
+    - pip
+    - hatch-jupyter-builder
+  run:
+    - python >=3.8
+    - jupyter_server >=1.6,<3
+    - jupyterlab >=4.0,<5
+    - pydantic >=1.0,<2.dev0
+    - openai >=0.26,<1.dev0
```
[the remark this answers is not in the document]
Comment (3coins) [reply recipes/jupyter-ai/meta.yaml:30]: @SylvainCorlay 
Thanks for reviewing. I am not sure how to be certain about this, but I could not find any dev releases for both of these with `conda search`. Given that info, should we update the upper bound for these packages?

**Comment by 3coins** [comment]
@SylvainCorlay 
Updated the dependencies based on your suggestion.

File: recipes/jupyter-ai/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SylvainCorlay) [reply recipes/jupyter-ai/meta.yaml]: Normally, conda-forge should not include pre-releases. (On a separate note, I also think this should not be needed for a PyPI release.)

**Comment by SylvainCorlay** [comment]
Thanks!

**Pull request merged**